### PR TITLE
Add Sonarcloud

### PR DESCRIPTION
**Pull request by joachimdekker**
This PR proposes to add Sonarcloud to the build pipeline. I have no idea how GitHub Actions work, so I just pasted the pipeline provided by Sonarcloud and added the dotnet commands. 

@QuakeEye, please review it 🙏

### CODE REVIEW

File: .github/workflows/pr-build.yml
Comment (SilasPeters) [inline .github/workflows/pr-build.yml:1]: Perhaps some more explanation of why we need certain tasks?

**Review by SilasPeters — CHANGES_REQUESTED** [review]

**Review by SilasPeters — DISMISSED** [review]

File: .github/workflows/pr-build.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    branches:
+      - main
+  pull_request:
+    types: [opened, synchronize, reopened]
+jobs:
+  build:
+    name: Build and analyze
+    runs-on: windows-latest
+    steps:
+      - name: Set up JDK 17
```
Comment (QuakeEye) [inline .github/workflows/pr-build.yml:13]: Would maybe help to specify in the name that this is installed for sonarcloud as mentioned previously

File: .github/workflows/pr-build.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        uses: actions/setup-java@v3
+        with:
+          java-version: 17
+          distribution: 'zulu' # Alternative distribution options are available.
```
Comment (QuakeEye) [inline .github/workflows/pr-build.yml]: I do not think we will change this, the comment can probably be removed

File: .github/workflows/pr-build.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
```
Comment (QuakeEye) [inline .github/workflows/pr-build.yml]: This is a github thing, I don't think specifying the fetch depth is even necessary at all, as we just want to retrieve everything in this pull request

File: Aplib.Core/Main.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
+using System;
+
+namespace Aplib.Core
+{
+    public class SonarTest
```
Comment (SSelinn) [inline Aplib.Core/Main.cs]: Change class name to used format. Same as filename. :D

File: Aplib.Core/Main.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        public void Main()
+        {
+            Console.WriteLine("Hello World!");
+            int i = 0;
+
+            while (i < 10)
+            {
+                Console.WriteLine(i);
+                i++;
+            }
```
Comment (SSelinn) [inline Aplib.Core/Main.cs]: Why did you not use a for-loop? Any reason to use a while-loop?

File: Aplib.Core/Main.cs [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+            Console.WriteLine("i is 1");
```
Comment (SSelinn) [inline Aplib.Core/Main.cs]: Isn't "i" a variable? Are you checking if i == 1 or are you stating it?

File: Aplib.Tests/Aplib.Tests.csproj [hunk 7]
```diff
@@ -0,0 +1,29 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
```
Comment (SSelinn) [inline Aplib.Tests/Aplib.Tests.csproj:4]: Was this not something else or was it 8?

File: Aplib.Tests/SonarTest.cs [hunk 8]
```diff
@@ -0,0 +1,13 @@
+namespace Aplib.Tests;
+
+public class SonarTest
```
Comment (SSelinn) [inline Aplib.Tests/SonarTest.cs:3]: Name format o.o

**Review by SSelinn — DISMISSED** [review]
:D

File: Aplib.Tests/Aplib.Tests.csproj (hunk 7 above)
Comment (joachimdekker) [reply Aplib.Tests/Aplib.Tests.csproj:4]: The testing project can be in .NET 8, as .NET 8 can import .NET standard 2.1 libraries and test them.

File: Aplib.Tests/SonarTest.cs (hunk 8 above)
Comment (joachimdekker) [reply Aplib.Tests/SonarTest.cs:3]: What do you mean? Also, what do you propose?

**Review by SilasPeters — DISMISSED** [review]
LGTM!

**Review by QuakeEye — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-build.yml (hunk 1 above)
Comment (joachimdekker) [reply .github/workflows/pr-build.yml:13]: Isn't that a bit redundant?

File: .github/workflows/pr-build.yml [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+          .\.sonar\scanner\dotnet-sonarscanner begin
+          /k:"team-zomsa_aplib.net" /o:"team-zomsa"
+          /d:sonar.token="${{ secrets.SONAR_TOKEN }}"
+          /d:sonar.host.url="https://sonarcloud.io"
+          /d:sonar.cs.vscoveragexml.reportsPaths=coverage.xml
+      - name: Build the project
+        shell: powershell
+        run: |
+          dotnet build --no-incremental
+
```
Comment (QuakeEye) [inline .github/workflows/pr-build.yml]:
```suggestion
```

🤨

**Review by QuakeEye — CHANGES_REQUESTED** [review]

File: Aplib.Tests/SonarTest.cs (hunk 8 above)
Comment (QuakeEye) [reply Aplib.Tests/SonarTest.cs:3]: This was due to the class name not being the same as the filename which had been solved, marking as resolved was alright.

**Review by QuakeEye — APPROVED** [review]
lgtm!

**Review by SilasPeters — APPROVED** [review]

**Pull request merged**